### PR TITLE
SHARD-2383 - Add unit tests for Utils

### PR DESCRIPTION
**Pull request by jintukumardas**
### **User Description**

image

### **PR Type**
Tests

___

### **Description**
- Add comprehensive unit tests for `Utils` module.

- Mock `Logger.mainLogger` to prevent errors.

- Test various utility functions including `getRandom`, `shuffleArray`, and `robustPromiseAll`.

- Validate functionality of `verifyMultiSigs` and `robustQuery`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Utils.test.ts</strong><dd><code>Add unit tests for Utils module</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/unit/src/Utils.test.ts

<li>Added unit tests for multiple utility functions.<br> <li> Mocked <code>Logger.mainLogger</code> to avoid errors during tests.<br> <li> Included tests for error handling and edge cases.<br> <li> Verified functionality of signature verification and query methods.

</details>

  </td>
  <td><a href="https://github.com/shardeum/archiver/pull/184/files#diff-de8cec9a94bf76d0ced21bc1cb056e615957f4cb47aa5d8fe1c2e821490e12b4">+569/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

**Comment by jintukumardas** [comment]
Skipped tests:

[Skipped Test](https://github.com/shardeum/archiver/pull/184/files#diff-de8cec9a94bf76d0ced21bc1cb056e615957f4cb47aa5d8fe1c2e821490e12b4R117-R119)
binarySearch returns -array length when element is not found. Not sure if this is expected outcome

[Skipped Test](https://github.com/shardeum/archiver/pull/184/files#diff-de8cec9a94bf76d0ced21bc1cb056e615957f4cb47aa5d8fe1c2e821490e12b4R91)
computeMedian returns incorrect median

[Skipped Test](https://github.com/shardeum/archiver/pull/184/files#diff-de8cec9a94bf76d0ced21bc1cb056e615957f4cb47aa5d8fe1c2e821490e12b4R196-R208)
verifyMultiSigs signs payload hash instead of payload; Ticket already exists

File: test/unit/src/Utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+      const [resolved, errors] = await Utils.robustPromiseAll(promises)
+      expect(resolved).toEqual([])
+      expect(errors).toHaveLength(2)
+    })
+  })
+
+  describe('computeMedian', () => {
+    it('should compute median for odd length array', () => {
+      expect(Utils.computeMedian([1, 2, 3])).toEqual(2.5)
+    })
```
[the remark this answers is not in the document]
Comment (jintukumardas) [reply test/unit/src/Utils.test.ts]: Skipped this test and added a comment. Need to fix the compute method to return correct output.

**Comment by jintukumardas** [comment]
> Skipped tests:
> 
> [Skipped Test](https://github.com/shardeum/archiver/pull/184/files#diff-de8cec9a94bf76d0ced21bc1cb056e615957f4cb47aa5d8fe1c2e821490e12b4R117-R119) binarySearch returns -array length when element is not found. Not sure if this is expected outcome
> 
> [Skipped Test](https://github.com/shardeum/archiver/pull/184/files#diff-de8cec9a94bf76d0ced21bc1cb056e615957f4cb47aa5d8fe1c2e821490e12b4R91) computeMedian returns incorrect median
> 
> [Skipped Test](https://github.com/shardeum/archiver/pull/184/files#diff-de8cec9a94bf76d0ced21bc1cb056e615957f4cb47aa5d8fe1c2e821490e12b4R196-R208) verifyMultiSigs signs payload hash instead of payload; Ticket already exists

I lost the master ticket link where we need to create sub-ticket for the failed tests
CC: @justin-shardeum
